src/main/java/com/conveyal/datatools/manager/controllers/api/FeedVersionController.java

```diff
@@ -272,17 +270,23 @@ private static FeedVersion publishToExternalResource (Request req, Response res)
         FeedVersion version = requestFeedVersion(req, "manage");
 
         // notify any extensions of the change
-        for(String resourceType : DataManager.feedResources.keySet()) {
-            DataManager.feedResources.get(resourceType).feedVersionCreated(version, null);
-        }
-        if (!DataManager.isExtensionEnabled("mtc")) {
-            // update published version ID on feed source
-            Persistence.feedSources.updateField(version.feedSourceId, "publishedVersionId", version.namespace);
-            return version;
-        } else {
-            // NOTE: If the MTC extension is enabled, the parent feed source's publishedVersionId will not be updated to the
-            // version's namespace until the FeedUpdater has successfully downloaded the feed from the share S3 bucket.
-            return Persistence.feedVersions.updateField(version.id, "processing", true);
+        try {
+            for (String resourceType : DataManager.feedResources.keySet()) {
+                DataManager.feedResources.get(resourceType).feedVersionCreated(version, null);
+            }
+            if (!DataManager.isExtensionEnabled("mtc")) {
+                // update published version ID on feed source
+                Persistence.feedSources.updateField(version.feedSourceId, "publishedVersionId", version.namespace);
+                return version;
+            } else {
+                // NOTE: If the MTC extension is enabled, the parent feed source's publishedVersionId will not be updated to the
+                // version's namespace until the FeedUpdater has successfully downloaded the feed from the share S3 bucket.
+                Date publishedDate = new Date();
+                return Persistence.feedVersions.updateField(version.id, "published", publishedDate);
+            }
+        } catch (Exception e) {
+            haltWithMessage(req, 400, "Could not publish feed.", e);
```

Comment:
This haltWithMessage should probably return a 500 error.  If the exception is actually due to bad user input, we should try to return a more helpful error message.